ui/pages/confirmations/hooks/alerts/useDomainMismatchAlert.ts

```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react';
+import {
+  isValidSIWEOrigin,
+  WrappedSIWERequest
+} from '@metamask/controller-utils';
+
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { SignatureRequestType } from '../../types/confirm';
+import { isSIWESignatureRequest } from '../../utils';
+import useCurrentConfirmation from '../useCurrentConfirmation';
+
+const useDomainMismatchAlerts = (): Alert[] => {
```

Comment:
Should we create some unit tests for this?

We can use `renderHookWithProvider` so it's easier to mock the state and test different scenarios like `is not message`, `is not SIWE`, `is domain valid` etc.

Reply:
Will do and then it will need to be commented out for now since that's how we're doing it in useCurrentConfirmation. A section there blocks the SIWE redesign unless it's commented out. 

(unsure why we went with this instead of an env flag)

Reply:
created tests here
[a07296d](https://github.com/MetaMask/metamask-extension/pull/25281/commits/a07296d10f5c85554a94a3933bc22fef046b7930)

Reply:
create tests [a07296d](https://github.com/MetaMask/metamask-extension/pull/25281/commits/a07296d10f5c85554a94a3933bc22fef046b7930) (added followup commits for cleanup)